src/transformation/utils/lua-ast.ts

```diff
@@ -143,11 +142,10 @@ export function createLocalOrExportedOrGlobalDeclaration(
             );
         }
     } else {
-        const insideFunction = findScope(context, ScopeType.Function) !== undefined;
-
-        if (context.isModule || getCurrentNamespace(context) || insideFunction || isVariableDeclaration) {
-            const scope = peekScope(context);
+        const scope = peekScope(context);
+        const isTopLevelVariable = scope.type === ScopeType.File;
 
+        if (context.isModule || getCurrentNamespace(context) || !isTopLevelVariable) {
```

Comment:
Have you checked if `getCurrentNamespace` is still required? I think it should push some scope?

Reply:
It does indeed push a block scope, removed the check